source/simulator/core.cc

```diff
@@ -2147,6 +2167,286 @@ namespace aspect
           break;
         }
 
+        case NonlinearSolver::Newton_Stokes:
+        {
+          if (parameters.free_surface_enabled)
+            free_surface->execute ();
+
+          double initial_temperature_residual = 0;
+          double initial_stokes_residual      = 0;
+          std::vector<double> initial_composition_residual (parameters.n_compositional_fields,0);
+
+          double initial_residual = 0;
+
+          double velocity_residual = 0;
+          double pressure_residual = 0;
+          double residual = 1;
+          double residual_old = 1;
```

Comment:
I have three different residuals, these two (they used to be called Newton_Stokes_residual) which I compute by caluculting the l2 norm of the Newton rhs (F) and an other one which is currently called the stokes residual and which is what the solve_stokes returns (normaly Ax-b, now with Newton Jdx-F). What are good names for these different variables?